agent/consul/authmethod/authmethods.go

```diff
@@ -185,6 +186,7 @@ func Types() []string {
 // ParseConfig parses the config block for a auth method.
 func ParseConfig(rawConfig map[string]interface{}, out interface{}) error {
 	decodeConf := &mapstructure.DecoderConfig{
+		DecodeHook:       decode.HookWeakDecodeFromSlice,
```

Comment:
I'm not sure where authmethod comes from. Maybe this is unnecessary? `HookWeakDecodeFromSlice` is more permissive, so it won't break anything, but it does expand the allowed values that can be decoded into a struct. If these values can't come from HCL, I will remove this line.

Reply:
AuthMethods cannot come from HCL. They are only ever sent to the API. The CLI takes options to build out the method. The CLI usage says that the format of the config "Must be JSON". So this is probably safe to remove.

Reply:
Excellent, I've reverted this change.